### ISSUE TITLE
Alvium 1800 Python multi-threading full resolution

### ISSUE THREAD

**Issue by patricknaughton01** [alliedvision/VimbaPython#120]
Hello,

I'm trying to stream two Allied Vision Alvium 1800 U-500c, 1/2.5" 5.0MP CS-Mount, USB 3.1 Color Cameras at their native 2592 x 1944 resolution in asynchronous mode using an Ubuntu 20.04 machine. Using the multi-threading example code (but changing the `FRAME_WIDTH` and `FRAME_HEIGHT` values), I am sometimes able to get both camera streams to show up on my computer, but other times one of the cameras will stop updating its frame, and some of the time no image will show up at all. I've noticed that if I reduce the width and height of the images by half, or only try to stream one camera, the stream works as expected. Is this an expected issue? What can I do to get both cameras simultaneously streaming to my computer at once? Thank you so much.

**Comment by Teresa-AlliedVision** [comment]
Hello Patrick,
this sounds like a bandwidth issue: The frames are not processed fast enough at the host, so the camera doesn't send any new frames until it gets confirmation that enough buffer is available for it to send the frames to. This leads usually to incomplete or dropped frames. 
You can up the number of buffers, just to check if that could be an issue, but usually you need to reduce the amount of data going through.
Try lowering the Device Link Throughput Limit until you can get a stream working. For finetuning, have a look at the frame statuses when they are processed in your program to check for incomplete frames. By tracking the frameIDs you can check for dropped frames.
Hope that helps,
Teresa

**Comment by patricknaughton01** [comment]
Thanks for getting back to me, I tried lowering the DeviceLinkThroughputLimit via the Vimba viewer to its minimum value of 98,375,000 and also using the vimba SDK programmatically but the behavior remains the same (by the way, is this value in bits per second or bytes per second?). Is it possible to somehow set this limit even lower to see if it will work? 

In terms of upping the number of buffers, is this referring to the `buffer_count` parameter passed to `start_streaming`? If so, we did try that and unfortunately did not see any change in behavior.

Also, something that's interesting is that when trying to stream both cameras it seems that the callback passed to `start_streaming` is never called for one of the cameras (a print at the beginning of the function is never output), not just that it's only receiving incomplete frames. The callback and prints will work for one camera at a time, but not both simultaneously. Is this expected?

**Comment by patricknaughton01** [comment]
Another thing I noticed is that if I set the cameras to Mono8 both cameras can be streamed simultaneously, but when I switch to Bgr8 I experience the above symptoms. Looking at the output of `usbtop`, it seems that the usb card of the computer is handling two 190 MB/s streams simultaneously. However, as I mentioned above, even when I set the throughput limit of each camera below this (at 98 MB/s) and try to stream in color, only one of the cameras can be streamed.
